go.mod

```diff
@@ -0,0 +1,26 @@
+module github.com/jaegertracing/jaeger-client-go/v3
+
+go 1.16
```

Comment:
If it's an initial 'go.mod' file, maybe it will be better to use the latest [go 1.17 ](service dependency analysis) version?

Reply:
well the go mod version just defines the minimum version that is needed to build the package. 
Is there any go1.17 feature that we directly need? 

But still, i could change it to go1.17 would be no problem